src/overlays/actors/ovl_En_Scopenuts/z_en_scopenuts.c

```diff
@@ -28,81 +45,773 @@ const ActorInit En_Scopenuts_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB078.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB1C8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB230.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB4DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB52C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB6D0.s")
+static Gfx* D_80BCCCDC[] = { gameplay_keep_DL_0527F0, gameplay_keep_DL_0528B0 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB90C.s")
+static Vec3f D_80BCCCE4 = { 0.0f, -0.5f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCB980.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCBA00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCBC60.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCBD28.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCBF0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCBFFC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC288.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC2AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC448.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/EnScopenuts_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/EnScopenuts_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/EnScopenuts_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC828.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC9CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/func_80BCC9E4.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 30, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Scopenuts/EnScopenuts_Draw.s")
+s16 func_80BCABF0(Path* path) {
+    Vec3s* sp34 = Lib_SegmentedToVirtual(path->points);
+    Vec3f sp28;
+    Vec3f sp1C;
+
+    Math_Vec3s_ToVec3f(&sp28, &sp34[0]);
+    Math_Vec3s_ToVec3f(&sp1C, &sp34[1]);
+    return Math_Vec3f_Yaw(&sp28, &sp1C);
+}
+
+void func_80BCAC40(EnScopenuts* this, GlobalContext* globalCtx) {
+    s16 i;
+    Vec3f sp60;
+    Vec3f sp54;
+
+    sp60 = this->actor.world.pos;
+    sp60.y += 100.0f;
+
+    for (i = 0; i < 36; i++) {
+        sp54.x = Rand_Centered() * 10.0f;
+        sp54.y = 2.0f * Rand_Centered();
+        sp54.z = Rand_Centered() * 10.0f;
+        EffectSsHahen_Spawn(globalCtx, &sp60, &sp54, &D_80BCCCE4, 0, 0x96, 1, 0x10, D_80BCCCDC[i & 1]);
```

Comment:
This would make a bit more sense with a `i % 2`. Maybe `i` is a u16?

Reply:
Doesn't work with either.